File: 2021W31/fienestar/pgr_77884.js
/*
    약수의 개수와 덧셈
    https://programmers.co.kr/learn/courses/30/lessons/77884
*/
function solution(left, right)
{
    let sum = 0
    for(; left <= right; ++left){
        let sqrt = Math.sqrt(left)
        if(sqrt == (sqrt | 0))
            sum -= left
        else
            sum += left
    }
    return sum;
}

File: 2024W48/brainer/pgr_77884.py
# 약수의 개수와 덧셈
# https://school.programmers.co.kr/learn/courses/30/lessons/77884

def solution(left, right):
    answer = 0
    
    for i in range(left, right + 1):
        divisors = 0
        for j in range(1, i + 1):
            if i % j == 0:
                divisors += 1
        if divisors % 2 == 0:
            answer += i
        else:
            answer -= i
    
    return answer
